src/components/CreateAnalysis/SelectionStudy.tsx

```diff
@@ -9,24 +9,29 @@ const SelectionStudy: React.FC<StudyInstance> = ({
   studyDescription,
   modality,
   createdDate,
+  setModelType,
+  index
 }) => {
   const { state: { createAnalysis: { selectedStudyUIDs, currSelectedStudyUID } }, dispatch } = useContext(AppContext);
   const imagesSelectedDict = selectedStudyUIDs[studyInstanceUID]
 
-  const selectThisStudy = () => {
+  const selectThisStudy = (index: number) => {
     dispatch({
       type: CreateAnalysisTypes.UpdateCurrSelectedStudyUID,
       payload: {
         studyUID: studyInstanceUID
       }
-    })
+    });
+    if (setModelType) {
+      setModelType(index);
+    }
   }
 
   const isSelected: boolean = !!imagesSelectedDict && Object.keys(imagesSelectedDict).length > 0;
   return (
     <div
       className={`SelectionStudy ${isSelected ? 'selected' : ''}`}
-      onClick={selectThisStudy}
+      onClick={() => {if (index !== undefined) selectThisStudy(index)}}
```

Comment:
Continuing from the comments made in CreateAnalysisDetail.tsx, this could be a potential change to make:
```suggestion
      onClick={() => {if (modality) selectThisStudy(modality)}}
```

Reply:
Agreeing on that point